components/operations-controller/internal/webhook/client.go

```diff
@@ -124,7 +124,12 @@ func (c *client) Do(ctx context.Context, request *Request) (*web_hook.Response,
 	isAsyncWebhook := webhook.Mode != nil && *webhook.Mode == graphql.WebhookModeAsync
 
 	if isLocationEmpty && isAsyncWebhook {
-		return nil, errors.New("missing location url after executing async webhook")
+		respBody, err := ioutil.ReadAll(resp.Body)
```

Comment:
The response body is already read in the `parseResponseObject`. We should close it in order to avoid resource leak. Closing the body there won't allow such second read.